packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -305,14 +292,12 @@ Static group authorization should perform as expected.`
                 resolver = this.resources.blankResolver(typeName, field.name.value)
             }
             const authExpression = this.authorizationExpressionOnSingleObject(rules, 'ctx.source')
-            if (protectPrivateFields) {
+            if (modelConfiguration.getName('level') === 'on') {
                 if (field.type.kind === Kind.NON_NULL_TYPE) {
                     throw new InvalidDirectiveError(`\nPer-field auth on the required field ${field.name.value} is not supported with subscriptions.
```

Comment:
nit: Error messages should not contain formatting like newlines, etc. We can fix it later.